File: nysol_python/mcmd/methods/convTex/db/smp/mdelnull.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mdelnull'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,quantity,amount
A,1,10
A,,20
B,1,15
B,3,
C,1,20
'''
db['idatas'].append(data)

data={}
data['name']='dat2.csv'
data['text']='''
customer,quantity,amount
A,1,10
A,,
B,1,15
B,3,
C,1,20
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``quantity`` と ``amount`` 項目がNULL値の行を削除する。
NULL値の行は ``oth.csv`` に出力する。
'''
script['sh_code']='''
mdelnull f=数量,金額 u=oth.csv i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mdelnull(f="quantity,amount", u="oth.csv", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='oth.csv,rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='NULL値の行を選択'
script['comment']='''
``r=True`` を指定することで、削除ではなく選択することになる。
'''
script['sh_code']='''
mdelnull f=数量,金額 -r i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mdelnull(f="quantity,amount", r=True, i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='キー項目でのNULL値の行の削除'
script['comment']='''
``k=`` を指定することで、集計キー単位で削除することになる。
以下では ``customer`` 項目を単位にして、 ``quantity`` と ``amount`` 項目にNULL値が一つでも含まれていれば削除する。
'''
script['sh_code']='''
mdelnull k=顧客 f=数量,金額 i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mdelnull(k="customer", f="quantity,amount", i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

script={}
script['title']='項目間AND条件の例'
script['comment']='''
``quantity`` と ``amount`` 項目の両方がNULL値の行を削除する。
'''
script['sh_code']='''
mdelnull f=数量,金額 -F i=dat2.csv o=rsl4.csv
'''
script['py_code']='''
nm.mdelnull(f="quantity,amount", F=True, i="dat2.csv", o="rsl4.csv").run()
'''
script['odatas']='rsl4.csv'
db['scripts'].append(script)

script={}
script['title']='レコード間AND条件の例'
script['comment']='''
``customer`` 項目を単位にして、 ``quantity`` 項目が全てNULL値の行を削除する。
'''
script['sh_code']='''
mdelnull k=顧客 f=数量 -R i=dat1.csv o=rsl5.csv
'''
script['py_code']='''
nm.mdelnull(k="customer", f="quantity", R=True, i="dat1.csv", o="rsl5.csv").run()
'''
script['odatas']='rsl5.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mdelnull.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer,Quantity,Amount
A,1,10
A,,20
B,1,15
B,3,
C,1,20
EOF
)}

File.open("dat2.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer,Quantity,Amount
A,1,10
A,,
B,1,15
B,3,
C,1,20
EOF
)}


############## Example 1
title="Basic Example"
comment=<<'EOF'
Remove records where \verb|Quantity| and \verb|Amount| contain null values. Save records with null values to a separate file \verb|oth.csv|.
EOF
scp=<<'EOF'
more dat1.csv
mdelnull f=Quantity,Amount u=oth.csv i=dat1.csv o=rsl1.csv
more rsl1.csv
more oth.csv
EOF
run(scp,title,comment)

############## Example 2
title="Select rows with NULL values"
comment=<<'EOF'
Select records with NULL values by specifying \verb|-r|.
EOF
scp=<<'EOF'
mdelnull f=Quantity,Amount -r i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Remove records with the same key if any record contains NULL values"
comment=<<'EOF'
Remove based on the aggregate key specified at \verb|k=|.
Remove records where \verb|Quantity| and \verb|Amount| contain null values for each customer.
EOF
scp=<<'EOF'
mdelnull k=Customer f=Quantity,Amount i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

############## Example 4
title="AND condition between fields"
comment=<<'EOF'
Remove the record where both \verb|Quantity| and \verb|Amount| fields contain null values.
EOF
scp=<<'EOF'
more dat2.csv
mdelnull f=Quantity,Amount -F i=dat2.csv o=rsl4.csv
more rsl4.csv
EOF
run(scp,title,comment)

############## Example 5
title="AND condition between records"
comment=<<'EOF'
Remove the \verb|Customer| record if all values in \verb|Quantity| is NULL.
EOF
scp=<<'EOF'
mdelnull k=Customer f=Quantity -R i=dat1.csv o=rsl5.csv
more rsl5.csv
EOF
run(scp,title,comment)
